cli/azd/cmd/auth_login.go

```diff
@@ -368,3 +392,30 @@ func (la *loginAction) login(ctx context.Context) error {
 
 	return nil
 }
+
+func parseUseDeviceCode(ctx context.Context, flag stringPtr, commandRunner exec.CommandRunner) (bool, error) {
+	var useDevCode bool
+
+	explicitDeviceCodeInput := flag.ptr != nil
+	if explicitDeviceCodeInput {
+		userInput, err := strconv.ParseBool(*flag.ptr)
+		if err != nil {
+			return false, fmt.Errorf("unexpected boolean input for use-device-code: %w", err)
+		}
+		// honor the value from the user input. No override.
+		return userInput, err
+	}
+
+	// Only try to override to device code for Codespaces. There's no point on checking if running on the browser all
+	// the time, as that's has a dependency on vscode.
```

Comment:
The comment confused me a little. What do you think about this rephrasing?

```suggestion
	// Detect cases where the browser isn't available for interactive auth, and we instead want to set `useDeviceCode` to be true by default
```